Alkiiro/refactor-tractor
# Fetch

#### What's this PR do?
This PR has merged Fosters' branch with the new fetch branch. It has replaced the data sets with the URL's to the database that Brittany set up. All four data sets are using fetch now and it's working as it did before 

#### Where should the reviewer start?
I would start by pulling it down and starting your server to make sure that it's all working on your machines! In addition, I would look at lines 15 to 18 and 72 to 101 in the index.js file. Lines 15 to 18 are pretty classic fetch requests that parses the data and then returns the array that we want. On lines 72 to 101, this is something that we did not discuss in class, which is Promise.all(). This is a function that ensures that all of our fetch requests have been completed before loading the page with all of the information. It runs once all of the fetch promises are done. 

#### How should this be manually tested?
I would pull it down and then make sure the server and everything works on your machines!

#### Any background context you want to provide?
Mostly about the Promise.all() that was discussed above. I will add a blog post below from MDN.

#### What are the relevant tickets?
We did not make an issue for fetch, but it is in the rubric!

#### Screenshots (if appropriate)
<img width="1552" alt="Screen Shot 2019-10-23 at 6 37 03 PM" src="https://user-images.githubusercontent.com/47511765/67444039-a01a8b00-f5c4-11e9-9207-2b0658a2f3ea.png">

#### Questions:
- Is there a blog post?
Promise.all() documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all 

- Does the knowledge base need an update?
We want to make sure that all parties know how Promise.all() works!

@brittanystoroz @khalidwilliams
